benchmark/Streamly/Benchmark/Data/Parser/ParserD.hs

```diff
@@ -63,6 +63,10 @@ benchIOSink value name f =
 drainWhile :: MonadThrow m => (a -> Bool) -> PR.Parser m a ()
 drainWhile p = PR.takeWhile p FL.drain
 
+{-# INLINE sliceBeginWith #-}
+sliceBeginWith :: MonadCatch m => Int -> SerialT m Int -> m()
+sliceBeginWith value = IP.parseD (PR.sliceBeginWith (>= value) FL.drain)
```

Comment:
We should write a `parseMany sliceBeginWith` benchmark in the Prelude/Split benchmarks. So that we can measure the performance of this in splitting a stream in chunks.